### DISCUSSION TITLE
How to sync 2 self-hosted instances?

### DISCUSSION THREAD

**Discussion by RobertusIT** [dani-garcia/vaultwarden#4259] [General]
Hi,

I need to sync  2 self-hosted instances, because if a server goes down, there is another one, that works.

I use docker compose in a virtual machine, on Proxmox.

**Comment by redpixy** [comment]
you could try hosting vaultwarden with mariadb or some other SQL. And point both your vaultwarden to that single DB

**Reply by RobertusIT** [reply]
Please can you provide this, with a tutorial?

**Reply by redpixy** [reply]
For setting up the mariadb you can refer to the wiki's FAQ under database- https://github.com/dani-garcia/vaultwarden/wiki
Details on how to setup is there including the supported DBs infos.

As for compose file, you just need to link both your vaultwarden instances to that single DB.

**Comment by stefan0xC** [comment]
> if a server goes down, there is another one, that works.

Have you thought about [backing up your vault](https://github.com/dani-garcia/vaultwarden/wiki/Backing-up-your-vault) instead of trying something that is not really intended?

**Reply by RobertusIT** [reply]
Thanks for that, I backup every day, because is a virtual machine in proxmox, but is useless, because I can't lost my password in this way, but if server goes down, I want another one that have the same data.

Is like a raid, we must have 2 server to have 100% uptime.

A better way is a feature that sync 2 self hosted server.

Or a way to use 3 server, oke of them with mariadb, and 2 server that are connected to the main server with Maria. But the main server with Mariadb, should be a server with some stuff that, if goes down,  another one, with the same db, come up and other servers with bitwarden, can reach this new server.

I'm clear?

**Reply by redpixy** [reply]
Well for the database side of things, you could simply run your db of choice in high availability as long as its supported on the db version you using. Most Enterprise dbs supports high availability functionality.

**Comment by bokkabonga** [comment]
I remember that this topic comes up every now and then([see](https://github.com/dani-garcia/vaultwarden/discussions/1943) or [here](https://github.com/dani-garcia/vaultwarden/discussions/2691)) over the past few years. But this project isn´t really meant to be used to sync 2 or more instances.
Connecting to the same DB might work, but I wouldn´t really count on it to not do some funky stuff every now and then.

**Comment by pethsontaliscale** [comment]
This is an old thread.
If you run VaultWarden in a docker container it's realy easy to set up a new instance.
The real important stuff is in the MariaDB database. And MariaDB you can set up with High Availability with sync between servers.

**Reply by axelsimon** [reply]
So you're suggesting a "hot spare": Vault Warden container A, backed by MariaDB instance A replicated to MariaDB instance B, and a VW container B backed by MariaDB B.
If VW container A and MariaDB instance A go down, you switch to VW container B.

Has this been tested by anyone?

**Comment by develroo** [comment]
In theory you could [run your stack on a swarm that is physically seperate with a VPN.](https://www.khueapps.com/blog/article/how-to-run-docker-swarm-with-vpn) Then the failover will happen automatically but you'd need tooling to swap domains if you access it from outside.
